src/v/pandaproxy/json/iobuf.h

```diff
@@ -58,4 +61,37 @@ class rjson_parse_impl<iobuf> {
     serialization_format _fmt;
 };
 
+template<>
+class rjson_serialize_impl<iobuf> {
+public:
+    explicit rjson_serialize_impl(serialization_format fmt)
+      : _fmt(fmt) {}
+
+    bool operator()(rapidjson::Writer<rapidjson::StringBuffer>& w, iobuf buf) {
+        switch (_fmt) {
+        case serialization_format::none:
+            [[fallthrough]];
+        case serialization_format::binary_v2:
+            return encode_base64(w, std::move(buf));
+        case serialization_format::unsupported:
+            [[fallthrough]];
+        default:
+            return false;
+        }
+    }
+
+    bool
+    encode_base64(rapidjson::Writer<rapidjson::StringBuffer>& w, iobuf buf) {
+        iobuf_parser parser(std::move(buf));
+        auto str = parser.read_string(parser.bytes_left());
```

Comment:
i think we can avoid this linearization here by using base64 stream encoding interface. it looks like it would be something like this:

```
struct base64_state state;
base64_stream_encode_init(&state);
auto in = iobuf::iterator_consumer(b.cbegin(), b.cend());
in.consume(b.size_bytes(), [&state, &out](const char* src, size_t sz) {
    base64_stream_encode(&state, src, sz, out...);
});
base64_stream_encode_final...
```

the libbase64.h file seemed to also provide some guidance on how big the output buffer needs to be. your use of the uninitialized sstring for that makes sense.

Reply:
Good catch. I'm going to need a custom `OutputBuffer` for `rapidjson` that wraps `iobuf` to prevent unfragmented linearization into an `ss::sstring`, or even better, an `ss::output_stream`, I'm hoping I don't have to re-imagine all of `rapidjson` to make it async.

Reply:
i don't think i'm following along with your comment, but maybe you're taking things a few steps further? what i wrote about should still be sync, so need to futurize anything, and even though the iobuf is fragmented, the output ssstring base64 encoding wouldn't be. but i haven't looked at this is a lot detail.

Reply:
Your comment about linearizing from the input buffer isn't too bad to fix. Having that stream into the writer (rather than the `ss::sstring`requires a little extra effort. I'll resubmit without that, for now, but I'll add a TODO.

Reply:
got it. right, i think we had the same thought: we build a bunch of base64 encoded strings when building the output, and each one has a linearization buffer sstring. in principle that could also be avoided, but it seemed like it would be a lot of reogranization.